### PR TITLE
feat: Python course lesson about saving data

### PR DESCRIPTION
**Pull request by honzajavorek**
TODO

- [x] intro etc.
- [x] export to CSV
- [x] export to JSON
- [x] fix CI errors
- [x] exercises
- [x] polish the texts
- [x] lesson closing

### CODE REVIEW

**Review by metalwarrior665 — COMMENTED** [review]
Can we add Crawlee extension later?

**Comment by honzajavorek** [comment]
What do you mean by Crawlee extension? Similar to the [JS course](https://docs.apify.com/academy/web-scraping-for-beginners/), I want to tackle crawling now and then show how this all can be simplified using Crawlee. Not sure yet at which point I'll introduce Crawlee exactly, in the JS course it takes 5 more lessons before Crawlee is mentioned. I'd like to keep the code simple enough so it's easy to follow and once it isn't, I'd bring Crawlee in.

**Comment by honzajavorek** [comment]
More TODO:

- [ ] fix filename= in code examples, should be title=
- [ ] fix order of imports, keep them in subsequent order as they're added throughout the course (now Decimal is suddenly first... could be alphabetically, but imho that just adds unnecessary cognitive load)

**Review by mnmkng — APPROVED** [review]

File: sources/academy/webscraping/scraping_basics_python/08_saving_data.md [hunk 1]
```diff
@@ -1,13 +1,246 @@
...
+response = httpx.get(url)
+response.raise_for_status()
+
+html_code = response.text
+soup = BeautifulSoup(html_code, "html.parser")
+
+# highlight-next-line
+data = []
+for product in soup.select(".product-item"):
+    title = product.select_one(".product-item__title").text.strip()
```
Comment (vdusek) [inline sources/academy/webscraping/scraping_basics_python/08_saving_data.md:45]: `select_one` can return also a `None`, but I believe it's fine for the purpose of the academy

**Review by vdusek — APPROVED** [review]
Good job :slightly_smiling_face:

**Comment by mnmkng** [comment]
@honzajavorek is this mergeable? The "TODO" in the original post scared me a bit. All the TODOs are checked, but still not sure if we can merge. Thanks!

**Comment by honzajavorek** [comment]
@mnmkng yup, it is! I would keep it as a draft otherwise.

File: sources/academy/webscraping/scraping_basics_python/08_saving_data.md (hunk 1 above)
Comment (honzajavorek) [reply sources/academy/webscraping/scraping_basics_python/08_saving_data.md:45]: yup, I thought about this one, but opting to have the code shorter and fail fast if something's wrong

**Comment by honzajavorek** [comment]
Regarding reviews, is it okay to merge now? I assign all four of you, taking Ondra and Lukas as optional. But now I have Vlada approving and Ondra approving, so...

Would it make sense if I always wait for @vdusek (or other pythonistas willing to step in) to check Python, and then perhaps approval from any of you three (Ondra / Lukas / Michał) would then mean it's good to go?

**Comment by mnmkng** [comment]
Certainly good to merge 👍🏻 but I agree that waiting for @vdusek to check the Python makes sense 😅

**Pull request merged**